doc/tutorials/Chapter_2/00-Basic_Logic.md

```diff
@@ -0,0 +1,534 @@
+      portC = basicLogic.signals
+          .firstWhere((element) => element.name == 'output_c')
+          .value
+          .toInt();
+      print('a: $i, b: $j c: $portC');
+    }
+  }
+}
+```
+
+You can find the executable code at [Basic Logic](./f_logic_gate_part_3.dart).
+
+Congratulations!!! You have successfully build your first gate! 
+
+### Exercise 3:
+
+1. Build OR or NOR or XOR gate using ROHD.
+
+# Constants
+
+In ROHD, constants can often be inferred by ROHD automatically, but can also be explicitly defined using `Const`, which extends `Logic`.
+
+```dart
+class ConstantLogic extends Module {
+  ConstantLogic() : super(name: 'ConstantLogic') {
+    // TODO(user): (Required) Paste your Logic initialization here.
+    // ------------------------------------------------------------
+    // Declare Constant
+    final x = Const(5, width: 16);
+    print('The value of constant x is: ${x.value.toInt()}');
+
+    // TODO(user): (Required) Declare your input and output port.
+    // ----------------------------------------------------------
+    // Add ports
+    final signal1 = addInput('const_x', x, width: x.width);
+  }
+}
+```
+
+### Exercise 4:
+
+1. Create a constant of value 10 and assign to a Logic input.
+
+# Bus Ranges and Swizzling
+
+In the previous module, we learned about the `width` property of `Logic`. Now, we can perform operations like slicing and swizzling on `Logic` values.
+
+We can access multi-bit buses using single bits, ranges, or by combining multiple signals. Additionally, we can use operations like slicing and swizzling on `Logic` values.
+
+```dart
+// TODO(user): (Optional) Change [YourModuleName] to your own module name.
+// -----------------------------------------------------------------------
+class RangeSwizzling extends Module {
+  // TODO(user): (Optional) Public attributes can register here.
+  // ------------------------------------------------------------
+  late final Logic a;
+  late final Logic b;
+  late final Logic c;
+  late final Logic d;
+  late final Logic e;
+  late final Logic f;
+
+  // TODO(user): (Optional) 'ModuleName' can change to your own module name.
+  // -----------------------------------------------------------------------
+  RangeSwizzling() : super(name: 'RangeSwizzling') {
+    // TODO(user): (Required) Paste your Logic initialization here.
+    // ------------------------------------------------------------
+    // Declare Constant
+    a = Logic(name: 'signal_a', width: 4);
+    b = Logic(name: 'signal_b', width: 8);
+    c = Const(7, width: 5);
+    d = Logic(name: 'signal_d');
+    e = Logic(name: 'signal_e', width: d.width + c.width + a.width);
+    f = Logic(name: 'signal_f', width: d.width + c.width + a.width);
+
+    // TODO(user): (Required) Declare your input and output port.
+    // ----------------------------------------------------------
+    // Add ports
+    final signal1 = addInput('a', a, width: a.width);
```

Comment:
Again, making all these inputs and outputs is not really relevant for this tutorial and makes the code look more complex and verbose than it needs to